gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -13,9 +14,53 @@ internal class GameConsoleWriter : TextWriter
         public override string NewLine => "\n";
         public override Encoding Encoding => Encoding.UTF8;
 
-#pragma warning disable CA2101
+        public GameConsoleWriter(ILua lua)
+        {
+            if (OperatingSystem.IsWindows())
+            {
+                Msg = Msg_Tier0;
+            }
+            else if (OperatingSystem.IsLinux())
+            {
+                lua.PushSpecial(SPECIAL_TABLES.SPECIAL_GLOB);
+                lua.GetField(-1, "game");
+                lua.GetField(-1, "IsDedicated");
+                lua.MCall(0, 1);
+                bool is_dedicated = lua.GetBool(-1);
+                lua.Pop(3);
+
+                if (is_dedicated)
+                {
+                    Msg = Msg_Tier0;
+                }
+                else
+                {
+                    Msg = Msg_Tier0_Client;
+                }
+            }
+            else if (OperatingSystem.IsMacOS())
+            {
+                Msg = Msg_Tier0_MacOS;
+            }
+            else Msg = (string msg) => { throw new PlatformNotSupportedException(); };
+        }
+
+#pragma warning disable CA2101 // workaround for https://github.com/dotnet/roslyn-analyzers/issues/5009
+
         [DllImport("tier0", CallingConvention = CallingConvention.Cdecl)]
-        private static extern void Msg([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+        private static extern void Msg_Tier0([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
```

Comment:
This is not working. Since function is not called `Msg` anymore, you have to manually specify `EntryPoint` in DllImport: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.entrypoint?view=net-5.0

Reply:
totally forgot about it